### ISSUE TITLE
Transformer predicts identical hidden states in validation

### ISSUE THREAD

**Issue by LeoZDong** [facebookresearch/detr#367]
I'm integrating DETR to another model and try to use DETR to perform bounding box prediction for 3d. I use PointNet (untrained) for feature extraction and pass it to some Conv1d layers to turn the feature into a sequence, used directly as input to the DETR transformer (so I bypass the backbone part). 

Since I have gt 3d bounding boxes, I thought this extension would be pretty straightforward. But I'm having trouble using DETR to overfit even 1 single example (the example has 5 bounding boxes). The predictions would start as almost identical bounding boxes for all steps and don't really change meaningfully as training progresses. In addition, if I turn off the dropout or test on the same single example in validation mode, I get identical bounding boxes for all steps. The predicted bounding box seems to try to bound the 3d thing in a _single_ box (i.e. it takes the boundary of all 5 gt boxes).

Is this "predicting same bounding boxes" problem expected? Why is dropout breaking the predictions (I looked and they are predicting identical bounding boxes because the decoder hidden states are identical)?

Thanks a lot!!

**Comment by LeoZDong** [comment]
Turns out that I was extracting my features incorrectly, which leads to DETR not able to differentiable / back-propagate between different steps. Closing this issue now.

**Comment by kritiyer** [comment]
@LeoZDong how did you create 3d bounding boxes in COCO format? Or did you have a custom data loader for DETR?

**Comment by LeoZDong** [comment]
@kritiyer I have a custom data loader

**Comment by mariiak2021** [comment]
Hi @LeoZDong I'm experiencing the same problem now in .eval() mode for the network. Could you provide some details of what was wrong specifically with feature maps in your case? Because I'm also trying to change the backbone from DETR to another model.
